### ISSUE TITLE
kaguyatc2isis - need to update for ingestion of reprocessed data

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#1764]
---

Author Name: **Tammy Becker** (Tammy Becker)

---

This was posted on the external isis-support board.
https://isis.astrogeology.usgs.gov/IsisSupport/index.php/topic,3732.0.html

Dear ISIS support,
the Kaguya TC team reprocessed all the data. Unfortunately, kaguyatc2isis does not work anymore as the label entries changed and the image data is now in uint format (formerly int). This is how I tested:
Code: [Select]

wget --content-disposition http://l2db.selene.darts.isas.jaxa.jp/dl/load_datafile.cgi?f=TCO_MAPm04_N09E006N06E009SC
tar xvf TCO_MAPm04_N09E006N06E009SC.sl2
kaguyatc2isis from=TCO_MAPm04_N09E006N06E009SC.img to=TCO_MAPm04_N09E006N06E009SC.cub

Maybe you can fix this sometime.
Regards
rschmidt

**Comment by lwellerastro** [comment]
I pulled the users specified file from the jaxa website (https://darts.isas.jaxa.jp/planet/pdap/selene/product_search.html) because the location he points to no longer exists.  

The ingestion failed, but in a different way than what he originally posted in 2013.  His file is in the older .sl2 format but what is made available via the website are pds-like (?) img and lbl files.

See /work/users/lweller/Isis3Tests/KaguyaTCCamera/MapData/ for the web data location (pdap_download.txt) and the downloaded files.

Here is the command and error message:

> kaguyatc2isis from=TCO_MAPm04_N09E006N06E009SC.img to=TCO_MAPm04_N09E006N06E009SC.cub
> 
>   Group = Error
>     Program = kaguyatc2isis
>     Code    = 1
>     Message = "Unable to read [DATA_SET_ID] from label file
>                [TCO_MAPm04_N09E006N06E009SC.img]"
>     File    = main.cpp
>     Line    = 36
>   End_Group
> 
>   Group = Error
>     Program = kaguyatc2isis
>     Code    = 1
>     Message = "PVL Keyword [DATA_SET_ID] does not exist in [Object = Root] in
>                file [TCO_MAPm04_N09E006N06E009SC.img]"
>     File    = PvlContainer.cpp
>     Line    = 84
>   End_Group

This error is similar to the one reported in [#3670 ](url)

**Comment by lwellerastro** [comment]
Note: the original issue (from 2013) was addressed at some point because we (USGS) created mosaicked TC map products around 2017 and it appears (after poking through someones directory and files) they ingested .sl2 based products from a local pds directory of likely specially delivered products from jaxa.

So the current problem is similar to the TC edr image ingestion problem where we based the program on a label format that was ultimately modified by jaxa before they served files to the public.

**Comment by jessemapel** [comment]
@lwellerastro It sounds like we need to re-examine all of the translations for the projected data similar to #3670

**Comment by AustinSanders** [comment]
This issue is due in part to the fact that the product uses a detached label, but the command targeted the .img file.   From the [kaguyatc2isis documentation](https://isis.astrogeology.usgs.gov/Application/presentation/Tabbed/kaguyatc2isis/kaguyatc2isis.html) the 'from' file "must contain the PDS labels."

Using the command
```
kaguyatc2isis from=TCO_MAPm04_N09E006N06E009SC.lbl to=out.cub
```
yields the following error which is resolved in #3713 
```
**ERROR** Input file [TCO_MAPm04_N09E006N06E009SC.lbl] does not appear to be a supported Kaguya Terrain Camera format. DATA_SET_ID is [SLN-L-TC-5-MORNING-MAP-V4.0]Valid formats include [TC_MAP, TCO_MAP, TC1_Level2B, SLN-L-TC-3-S-LEVEL2B0-V1.0].
```
---
In short, this did require a quick change to the code, but it also requires a change to the command.
